### PR TITLE
feat(fiori-gen): adds new module @sap-ux/fiori-generator-shared

### PR DESCRIPTION
**Pull request by cianmSAP**
#1850

Adds new module @sap-ux/fiori-generator-shared

### CODE REVIEW

File: packages/fiori-generator-shared/src/cap/index.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        db: 'db/',
+        srv: 'srv/'
+    };
+
+    const cdsrcPath = join(`${capProjectPath}/.cdsrc.json`);
+    const packageJsonPath = join(`${capProjectPath}/package.json`);
+
+    try {
+        for (const folder of Object.keys(capPaths)) {
+            if (!existsSync(join(`${capProjectPath}/${folder}`))) {
```
Comment (devinea) [inline packages/fiori-generator-shared/src/cap/index.ts:24]: Don't expect we should check for folder existence here. Just continue merging configs in order.

**Review by devinea — COMMENTED** [review]
New module setup looks good. 
Changeset ✅ 

one small comment.

**Review by devinea — APPROVED** [review]
Thanks for the updates.

File: packages/fiori-generator-shared/src/cap/index.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import { existsSync, readFileSync } from 'fs';
```
Comment (IainSAP) [inline packages/fiori-generator-shared/src/cap/index.ts:1]: Can we use fs promises instead of sync versions?

Comment (devinea) [reply packages/fiori-generator-shared/src/cap/index.ts:1]: This function is to be sync only, right?
maybe it should be called `getCapFolderPathsSync`

File: packages/fiori-generator-shared/test/cap/index.test.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { getCapFolderPaths } from '../../src/cap';
 import { join } from 'path';
 
 describe('getCapFolderPaths', () => {
+    jest.setTimeout(3000000);
```
Comment (IainSAP) [inline packages/fiori-generator-shared/test/cap/index.test.ts:6]: Should be removed?

Comment (cianmSAP) [reply packages/fiori-generator-shared/test/cap/index.test.ts]: Sorry merged to early by mistake

**Review by devinea — APPROVED** [review]
Re-approving

**Review by IainSAP — COMMENTED** [review]
Thanks @cianmSAP 

- Cov is good

**Review by IainSAP — APPROVED** [review]

**Pull request merged**